solutions/devsprint-riccardo-1/OnboardingChallengeKit/Sources/OnboardingChallengeKit/Screens/List/ListViewController.swift

```diff
@@ -3,34 +3,74 @@
 import UIKit
 
 public final class ListViewController: UIViewController {
-  private lazy var listView: ListView = {
-    return ListView()
-  }()
-
-  private let service = Service()
-
-  public init() {
-    super.init(nibName: nil, bundle: nil)
-  }
-
-  required init?(coder _: NSCoder) {
-    fatalError("init(coder:) has not been implemented")
-  }
-
-  override public func loadView() {
-    self.view = self.listView
-  }
-
-  override public func viewDidLoad() {
-    super.viewDidLoad()
-    self.fetchList()
-  }
-
-  private func fetchList() {
-    self.service.fetchList { items in
-      let configuration = ListViewConfiguration(listItems: items)
+    
+    // MARK: Outlets
+    private lazy var listView: ListView = {
+        return ListView()
+    }()
+    private lazy var searchController: UISearchController = {
+        let searchController = UISearchController(searchResultsController: nil)
+        searchController.delegate = self
+        searchController.searchBar.delegate = self
+        searchController.searchResultsUpdater = self
+        searchController.searchBar.placeholder = "Type a GitHub user name"
+        searchController.hidesNavigationBarDuringPresentation = false
+        return searchController
+    }()
+    private let service = Service()
+    
+    // MARK: Initialization
+    public init() {
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override public func loadView() {
+        self.view = self.listView
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        self.setupUI()
+        self.bindEvents()
+        self.fetchList()
+    }
+    
+    // MARK: Actions
+    //insert action to settingsViewController here
+    
+    // MARK: Methods
+    private func setupUI() {
+        title = "Repositories"
+        setupNavigationBar()
+    }
+    
+    private func setupNavigationBar() {
+        navigationController?.navigationBar.backgroundColor = .systemGray6
```

Comment:
A better alternative could be the `.secondarySystemBackground` color. In this way, if Apple will ever change the background color, the app will automatically adapt.